yarn.lock

```diff
@@ -21739,6 +21739,18 @@ object-path-immutable@^3.1.1:
   dependencies:
     is-plain-object "3.0.0"
 
+object-prototype-functions@^1.2.0:
```

Comment:
I thought it probably wouldn't change in the future (so far all these functions have been there from day one). And I wanted to only copy the functions anyway - not the properties - so I thought this would be easier and safer as I could also add tests for each of them to ensure they still worked as expected. And then I also didn't have to worry about an already polluted prototype.

But I'm open to changing this if you think it's best.

Reply:
> But I'm open to changing this if you think it's best.

I'm torn... If `object-prototype-functions` was only used in the context of Node.js itself, it'd be easy enough to write a test to ensure that we've taken every property into consideration and there aren't new ones. However, because `object-prototype-functions` appears to have functions which only exist in certain browsers, it becomes much harder to write tests to ensure this. 

I feel like we should either make `object-prototype-functions` Node.js specific and write the tests; or change it to be dynamic and use `Object.getOwnPropertyNames(Object.prototype)`

Reply:
This module started out only containing the original ECMAScript function names - hence why it exposed a hardcoded list. I then later decided to also include non-spec'ed names, and finally a convenience-list for Node.js specifically.

For the Node.js use-case, you're right we could probably just rely on something akin to `Object.getOwnPropertyNames(Object.prototype)`, but then we could do that directly in the `object-prototype` and wouldn't need a separate module like `object-prototype-functions`.

I would still like the tests that I have in the `object-prototype` module. I think they are valuable. The way I would normally deal with the case that a new function might be added to `Object.prototype` later, is to have a test that checks that `Object.getOwnPropertyNames()` returns what I expect. The moment it doesn't I fail the test so that I know the module doesn't cover everything.

But I think that the chances of a new function being added to `Object.prototype` is extremely low. And if it does (a) most devs will use a fallback if it's not present or (b) the Kibana tests will most likely fail.